### PR TITLE
Disallow sensitive trait on members

### PR DESCRIPTION
**Pull request by gosar**
Allowing @sensitive trait on members can lead to insecure usage
patterns, like forgetting to put @sensitive on ALL members that target
the same shape, which can lead to sensitive data not getting handled
properly, e.g., not getting redacted from logs, etc.

Instead marking the target shape as sensitive prevents modeling
mistakes by ensuring every reference to data types that are inherently
sensitive are always considered sensitive.

TODO: Doc updates. Currently `idl-2.0` branch's `docs` still references version 1.0 mostly. Will update docs separately later.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-jsonschema/src/test/java/software/amazon/smithy/jsonschema/JsonSchemaConverterTest.java [hunk 1]
```diff
@@ -551,6 +552,25 @@ public void supportsMapPatternPropertiesWithDefaultPattern() {
         assertThat(schema.getPatternProperties().get(".+").getType().get(), equalTo("string"));
     }
 
+    @Test
+    public void sensitiveTraitHasNotImpact() {
```
Comment (kstich) [inline smithy-jsonschema/src/test/java/software/amazon/smithy/jsonschema/JsonSchemaConverterTest.java:556]: One of these strings shouldn't have the trait applied.

**Review by kstich — CHANGES_REQUESTED** [review]

**Comment by mtdowling** [comment]
The spec probably needs to be updated to reflect this change. The selector shouldn't allow members for sensitive. Our implementation of Smithy will need to allow it since we support v1 and v2 simultaneously, but the extra validation we're doing will cover us to not actually allow it on members.

File: smithy-jsonschema/src/test/java/software/amazon/smithy/jsonschema/JsonSchemaConverterTest.java (hunk 1 above)
Comment (gosar) [reply smithy-jsonschema/src/test/java/software/amazon/smithy/jsonschema/JsonSchemaConverterTest.java:556]: haha, copy paste error. fixing test name too.

File: docs/source/1.0/spec/core/traits/documentation-traits.rst.template [hunk 2]
```diff
@@ -391,9 +391,9 @@ Summary
     Indicates that the data stored in the shape or member is sensitive
```
Comment (kstich) [inline docs/source/1.0/spec/core/traits/documentation-traits.rst.template:391]: rm "or member"

Comment (gosar) [reply docs/source/1.0/spec/core/traits/documentation-traits.rst.template]: fixed

**Review by kstich — APPROVED** [review]

**Review by mtdowling — APPROVED** [review]

**Pull request merged**